ledger_device_sdk/src/hash.rs

```diff
@@ -37,7 +37,7 @@ pub trait HashInit: Sized {
     fn get_size(&mut self) -> usize {
         unsafe { cx_hash_get_size(self.as_ctx()) }
     }
-    fn hash(mut self, input: &[u8], output: &mut [u8]) -> Result<(), HashError> {
+    fn hash(&mut self, input: &[u8], output: &mut [u8]) -> Result<(), HashError> {
```

Comment:
When calling `hash()`, the context `self` should be consumed to prevent any further usage later by the caller (same as `finalize()`): that why it is passed as a value and not as a ref The only use case where the context is reused is `update()`. I agree that it prevents a copy to be performed however...

Reply:
To be consistent, if we keep this change, it shall be also performed for `finalize()`

Reply:
Might be solved by https://github.com/LedgerHQ/ledger-device-rust-sdk/pull/143#discussion_r1529664279 ?